base/bootstrap.go

```diff
@@ -150,23 +150,53 @@ func NewCouchbaseCluster(ctx context.Context, server, username, password,
 	x509CertPath, x509KeyPath, caCertPath string,
 	forcePerBucketAuth bool, perBucketCreds PerBucketCredentialsConfig,
 	tlsSkipVerify *bool, useXattrConfig *bool, bucketMode BucketConnectionMode) (*CouchbaseCluster, error) {
+	opts := newCouchbaseClusterOpts{
+		server:             server,
+		username:           username,
+		password:           password,
+		x509CertPath:       x509CertPath,
+		x509KeyPath:        x509KeyPath,
+		caCertPath:         caCertPath,
+		forcePerBucketAuth: forcePerBucketAuth,
+		perBucketCreds:     perBucketCreds,
+		tlsSkipVerify:      tlsSkipVerify,
+		useXattrConfig:     useXattrConfig,
+		bucketMode:         bucketMode,
+	}
+	return NewCouchbaseClusterWithOpts(ctx, opts)
+}
+
+type newCouchbaseClusterOpts struct {
+	server             string
+	username           string
+	password           string
+	x509CertPath       string
+	x509KeyPath        string
+	caCertPath         string
+	forcePerBucketAuth bool
+	perBucketCreds     PerBucketCredentialsConfig
+	tlsSkipVerify      *bool
+	useXattrConfig     *bool
```

Comment:
nit: Given that this should either be true or false here and false is the default, do you think that this should just be concrete bool, and not pointer? 

I feel the same about tlsSkipVerify but that actually would have a downstream side effect of changing behavior of `GoCBv2SecurityConfig`

Reply:
Done, but left tlsSkipVerify because I want to limit code change down through to `GoCBv2SecurityConfig`